Day_26_27/miracleDay_26_27.scilla

```diff
@@ -0,0 +1,139 @@
+scilla_version 0
+
+import BoolUtils
+
+library MyContractLib
+
+
+(********* Part -5 Global Assignment "A" *************)
+
+type Question=
+  | Question of String ByStr20
+
+let one_msg =
+fun (msg : Message) =>
+let nil_msg = Nil {Message} in
+  Cons {Message} msg nil_msg
+
+let reward = Uint128 10000000000000 (reward 10 Zil)
+let one = Int32 1
+let not_an_oracle = Int32 1
+let wrong_ans = Int32 2
+let right_ans = Int32 3
+let no_ans = Int32 4
+let no_anss = Int32 5
+let answered = Int32 6
+let oracle_answered = Int32 7
+let no_oracle_ans = Int32 8
+
+
+contract Miracle
+(oracle: ByStr20)
+
+field currentquestionId : Int32 = Int32 0
+
+
+
+field finalOutput : Map Int32 String = Emp Int32 String
+field userVsAnswer : Map ByStr20 String = Emp ByStr20 String
+
+(********* Part -5 Global Assignment "B" *************)
+
+field questionMapping : Map Int32 Question = Emp Int32 Question 
+
+
+(********* Part -5 Global Assignment "C" *************)
+
+transition addQuestion (description : String)
+  q <- currentquestionId;
+  q = builtin add q one;
+  currentquestionId := q;
+  
+  sndr = _sender;
+  desc = description;
+  
+  qinfo = Question desc _sender;
+
+  questionMapping[q] := qinfo
+
+end
+
+
+transition userAnswer (qid: Int32, ans: String)
+
+  anss = ans;
+  accept;
+  userVsAnswer[_sender] := anss
```

Comment:
Why extra variable *anss* is used ?
You could directly use **ans**

Reply:
removed the anss and used ans .

> 
> 
> Why extra variable _anss_ is used ?
> You could directly use **ans**

removed the anss and used ans .